src/main/java/com/google/sps/data/User.java

```diff
@@ -14,12 +14,66 @@
 
 package com.google.sps.data;
 
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.QueryResultList;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+
 public final class User {
   private String id;
   private boolean isMaintainer = false;
+  private String userEmail;
 
-  public User(String id, boolean isMaintainer) {
+  public User(String id, boolean isMaintainer, String userEmail) {
     this.id = id;
     this.isMaintainer = isMaintainer;
+    this.userEmail = userEmail;
+  }
+
+  public boolean getMaintainerStatus() {
+    return this.isMaintainer;
+  }
+
+  public static User getUserFromDatastore(String userId) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Filter queryFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+    Query query = new Query("User").setFilter(queryFilter);
+      
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+    PreparedQuery preparedQuery = datastore.prepare(query);
+    QueryResultList<Entity> userResult = preparedQuery.asQueryResultList(fetchOptions);
+
+    boolean isMaintainer = false;
+    String userEmail = "";
+
+    if (userResult.size() < 1) {
+      return null;
+    }
```

Comment:
Do you have enough information to still return a valid user here? (Eg if the query failed, you know they aren't a maintainer)

Reply:
Hm, I'm not sure if I'm understanding this comment. The maintainer status doesn't really matter because this method is just getting a user with a string ID.